jjb/include/pipelines/qpc-tests-pipeline.groovy

```diff
@@ -0,0 +1,529 @@
+        -v /tmp:/tmp \
+        -v /home/jenkins/.ssh:/home/jenkins/.ssh \\
+        -v \${{PWD}}/log:/var/log \\
+        -i ${{image_name}}
+    """.stripIndent()
+
+    sh '''\
+    for i in {{1..30}}; do
+        SERVER_ID="$(curl -ks https://localhost:9443/api/v1/status/ | grep server_id || true)"
+
+        if [ "${{SERVER_ID}}" ]; then
+            break
+        fi
+
+        if [ $i -eq 30 ]; then
+            echo "Server took too long to start"
+            exit 1
+        fi
+
+        sleep 1
+    done
+    '''.stripIndent()
+}}
+
+
+
+//////////////////////////////
+//// Get Builds Functions ////
+//////////////////////////////
+def getQuipucords() {{
+    // get QPC
+    echo 'getQuipucords'
+    if ('{release}' == 'master') {{
+        getMasterQPC()
+    }} else {{
+        getReleasedQPC()
+    }}
+}}
+
+
+def getQuipucordsBuild() {{
+    // Grabs the quipucords build
+    echo "getQuipucords: Copying Latest build artifact..."
+    copyArtifacts filter: 'quipucords.*.tar.gz', fingerprintArtifacts: true, projectName: 'qpc-testing-build', selector: lastCompleted()
+    copyArtifacts filter: 'quipucords.*.install.tar.gz', fingerprintArtifacts: true, projectName: 'qpc-testing-build', selector: lastCompleted()
+    copyArtifacts filter: 'postgres.*.tar.gz', fingerprintArtifacts: true, projectName: 'qpc-testing-build', selector: lastCompleted()
+}}
+
+
+def getMasterQPC() {{
+    // Grabs the master qpc builds and sets it up
+    echo 'getMasterQPC'
+    def qpc_version = getQPCVersion()
+    getQuipucordsBuild()
+
+    sh 'ls -lah'
+    echo 'Extract the installer script'
+    sh """\
+    echo ${{qpc_version}}
+    tar -xvzf quipucords.${{qpc_version}}.install.tar.gz
+    """.stripIndent()
+
+    echo 'Copy container to installer packages directory'
+    sh '''\
+    mkdir -p install/packages
+    '''.stripIndent()
+
+}}
+
+
+def getReleasedQPC() {{
+    echo 'Get released QPC'
+    // Clean up QPC package if here...
+    sh '''\
+    ls -l
+    if [ -f quipucords.*tar.gz ]; then
+        rm quipucords.*.tar.gz
+    fi
+    '''.stripIndent()
+
+    // Pulls down Released Container
+    echo "load docker container from tarball"
+    sh "curl -k -O -sSL https://github.com/quipucords/quipucords/releases/download/{release}/quipucords.{release}.tar.gz"
+    // Pulls down Released Install
+    sh "curl -k -O -sSL https://github.com/quipucords/quipucords/releases/download/{release}/quipucords.{release}.install.tar.gz"
+
+    echo "extract the installer into ${{WORKSPACE}}/install"
+
+    sh "tar -xvzf ${{WORKSPACE}}/quipucords.{release}.install.tar.gz"
+}}
+
+
+
+///////////////////////////
+//// Install Functions ////
+///////////////////////////
+def installQPC(distro) {{
+    // Install QPC
+    if ('{install_type}' == 'nosupervisord') {{
+        installQPCNoSupervisorD()
+    }} else if ('{install_type}' == 'container') {{
+        containerInstall(distro)
+    }} else {{
+        defaultInstall()
+    }}
+    // Install the client
```

Comment:
do you mean to be calling your `installQpcClient` function here?

Reply:
Nope, just a stray comment. I was going to put it there but then realized I don't _always_ want to install the client on nodes. So, now I have it so that `installQpcClient` is explicitly called from the stage when needed/desired.